### ISSUE TITLE
adding `lme4` as a dependency

### ISSUE THREAD

**Issue by IndrajeetPatil** [easystats/insight#238]
Since we expect that other developers will probably soon start relying on `insight`, I think we need to add `lme4` as a dependency.

This is a problem for other developers because `lme4` is currently in `Suggests`, which means if a developer uses `parameters` or other package in their own package and don't explicitly add `lme4` to `Imports` (which is logical, since they didn't use any function from `lme4`), the CRAN check is going to fail because it can't find `lme4` anywhere.

For example, I run into this very problem:
I use `parameters::model_parameters` to extract parameters from `BayesFactor` for ANOVA models in `tidyBF`.

`parameters::model_parameters.BayesFactor` at some point uses this code

https://github.com/easystats/insight/blob/5bce6619d0270d5bad89dff6ea919ffc235b94b3/R/helper_functions.R#L162-L166 

And then the check fails-

```r
-- 2. Error: with subject.id (@test-bf_oneway_anova.R#310)  --------------------
To use this function, please install package 'lme4'.
Backtrace:
  1. tidyBF::bf_oneway_anova(...)
  2. tidyBF::bf_expr(...)
  3. tidyBF::bf_extractor(...)
 12. parameters:::model_parameters.BFBayesFactor(...)
 14. insight:::get_parameters.BFBayesFactor(model) C:/Users/inp099/Documents/insight/R/get_parameters.R:55:2
 16. insight:::find_parameters.BFBayesFactor(...) C:/Users/inp099/Documents/insight/R/find_parameters.R:50:2
 17. insight::find_variables(x, effects = "all") C:/Users/inp099/Documents/insight/R/find_parameters.R:866:4
 18. insight::find_predictors(...) C:/Users/inp099/Documents/insight/R/find_variables.R:64:2
 19. insight:::.prepare_predictors(x, f, elements) C:/Users/inp099/Documents/insight/R/find_predictors.R:55:4
 20. insight:::.get_group_factor(x, f[["random"]]) C:/Users/inp099/Documents/insight/R/find_predictors.R:139:4
 21. insight:::.get_model_random(f, split_nested = TRUE, x) C:/Users/inp099/Documents/insight/R/helper_functions.R:227:4
``` 

I can fool the `R CMD CHECK` by just importing some functions from `lme4` and adding it to `Imports`, but this will definitely become a hurdle for other developers and an excuse to not be excited at the prospect of using `easystats` in their development workflows.

What do you think?

**Comment by strengejacke** [comment]
Isn't it enough to add _lme4_ to the suggested packaged?

**Comment by strengejacke** [comment]
I _think_ you can add lme4 to suggests, and then add `if (require("lme4")) <do stuff>` to your tests.

**Comment by IndrajeetPatil** [comment]
Yeah, sure, that’s another solution. 

I personally don’t like the approach of conditionally suggesting packages but I totally see the value of doing that for packages with massive functionality (like parameters), not all of which will be used by all users. 

If we want to continue with the current approach, then we need to add a note for other developers about this issue. Does that sound reasonable?

**Comment by strengejacke** [comment]
>  then we need to add a note for other developers about this issue.

I'm not sure in which way. What do you have in mind? I don't see the need for such a note, because this is nothing "special" per se.

**Comment by strengejacke** [comment]
@mattansb @DominiqueMakowski opinions?

**Comment by mattansb** [comment]
Why on earth does `insight::find_predictors.BFBayesFactor` need `lme4`?
This seems like the real issue here...

1. BayesFactor formula's don't ever have bars.
2. (In general, not all bars denote random effects (in some packages they are the offset or other...))
3. `lme4::findbars` seems simple enough that we can "port" it.

**Comment by DominiqueMakowski** [comment]
Yeah, using suggest is pretty standard in R I think. Though we could maybe add a sentence or two about it in https://github.com/easystats/easystats#dependencies if need be

**Comment by strengejacke** [comment]
There is no `insight::find_predictors.BFBayesFactor()` method...

**Comment by strengejacke** [comment]
`find_predictors()` works for all models, calling some internal methods, one of which uses `lme4::findbars()` on the model-formula. This is maybe not required for BFBayesFactor objects, however, since this sub-function just checks a _formula_ in general for random effects (no matter from which model), BFBayesFactor objects run through this function. We indeed might to "port" findbars internally for insight.

**Comment by IndrajeetPatil** [comment]
I also vote for porting!

**Comment by DominiqueMakowski** [comment]
> We indeed might to "port" findbars internally for insight.

we can indeed shamelessly steal it 🙄

**Comment by IndrajeetPatil** [comment]
Partly addressed in 0f7399f956165bfe2312caa459fd5ccd4cfd85c3

Once we have an internal implementation of `.nobars`, I will close this.
